fdbsyncd/fdbsync.cpp

```diff
@@ -0,0 +1,603 @@
+        else
+            info.op_type = FDB_OPER_DEL ;
+
+        SWSS_LOG_NOTICE("FDBSYNCD STATE FDB updates key=%s, operation=%s\n", key.c_str(), op.c_str());
+        
+        for (auto i : kfvFieldsValues(entry))
+        {
+            SWSS_LOG_INFO(" FDBSYNCD STATE FDB updates : "
+            "FvFiels %s, FvValues: %s \n", fvField(i).c_str(), fvValue(i).c_str());
+
+            if(fvField(i) == "port")
+            {
+                memcpy(info.port_name, fvValue(i).c_str(), fvValue(i).length());
+            }
+
+            if(fvField(i) == "type")
+            {
+                if(fvValue(i) == "dynamic")
+                    info.type = FDB_TYPE_DYNAMIC;
+                else if (fvValue(i) == "static")
+                    info.type = FDB_TYPE_STATIC;
+            }
+        }
+        macUpdateStateDB(&info);
+    }
+}
+
+void FdbSync::macAddSrcDB(struct fdb_info *info)
+{
+    string key = "Vlan" + to_string(info->vid) + ":" + info->mac;
+    fdbmac[key].port_name = info->port_name;
+    fdbmac[key].type      = info->type;
+
+    return;
+}
+
+bool FdbSync::macCheckSrcDB(struct fdb_info *info)
+{
+    string key = "Vlan" + to_string(info->vid) + ":" + info->mac;
+    if (fdbmac.find(key) != fdbmac.end())
+    {
+        SWSS_LOG_NOTICE("DEL_KEY %s ", key.c_str());
+        return true;
+    }
+
+    return false;
+}
+
+void FdbSync::macDelVxlanEntry(string auxkey, struct fdb_info *info)
+{
+    std::string vtep = m_mac[auxkey].vtep;
+
+    const std::string cmds = std::string("")
+        + " bridge fdb del " + info->mac + " dev " 
+        + m_mac[auxkey].ifname + " dst " + vtep + " vlan " + to_string(info->vid);
+
+    std::string res;
+    int ret = swss::exec(cmds, res);
+    if (ret != 0)
+    {
+        SWSS_LOG_INFO("Failed cmd:%s, res=%s, ret=%d", cmds.c_str(), res.c_str(), ret);
+    }
+
+    SWSS_LOG_NOTICE("Success cmd:%s, res=%s, ret=%d", cmds.c_str(), res.c_str(), ret);
+
+    return;
+}
+
+void FdbSync::macUpdateStateDB(struct fdb_info *info)
+{
+    char *op;
+    char *type;
+    string port_name = "";
+    string key = "Vlan" + to_string(info->vid) + ":" + info->mac;
+	short fdb_type;    /*dynamic or static*/
+
+    if (info->op_type == FDB_OPER_ADD)
+    {
+        macAddSrcDB(info);
+        op = "add";
+        port_name+= info->port_name;
+        fdb_type = info->type;
+        /* Check if this vlan+key is also learned by vxlan neighbor then delete learned on */
+        if (m_mac.find(key) != m_mac.end())
+        {
+            op = "replace";
+            macDelVxlanEntry(key, info);
+            SWSS_LOG_INFO("Local learn event deleting from VXLAN table DEL_KEY %s", key.c_str());
+            macDelVxlan(key);
+        }
+    }
+    else
+    {
+        op = "del";
+        if (macCheckSrcDB(info) == false)
+        {
+            return;
+        }
+		port_name+= fdbmac[key].port_name;
+		fdb_type = fdbmac[key].type;
+		fdbmac.erase(key);
+    }
+
+    if (fdb_type == FDB_TYPE_DYNAMIC)

```

Comment:
There is a question about the bridge fdb command, whether the remote MAC need to have `extern_learn` field

Reply:
Thanks for review, Yes extern_learn flag need to be set but currently exec command doesn't has extern_learn support, Hence it is not set.